<!-- Run rtabmap_ros -->
<launch>
    <arg name="camera1_name"     default="d435i"/>
    <arg name="camera2_name"     default="t265"/>
    <arg name="db_folder"        default="/media/$(env USER)/NEPTUNE2/slam_data" />
    <arg name="database_name"    default="surge_outdoor_old_depth.db" />

    <arg name="localization"     default="false"/>

    <!-- Uncompress images for stereo_image_rect -->
    <node name="republish_left" type="republish" pkg="image_transport" args="compressed in:=/$(arg camera1_name)/infra1/image_rect_raw raw out:=/$(arg camera1_name)/infra1/image_rect_raw" />
    <node name="republish_right" type="republish" pkg="image_transport" args="compressed in:=/$(arg camera1_name)/infra2/image_rect_raw raw out:=/$(arg camera1_name)/infra2/image_rect_raw" />

    <!-- Uncompress images for color and depth -->
    <!-- <node name="republish_color" type="republish" pkg="image_transport" args="compressed in:=/$(arg camera1_name)/color/image_raw raw out:=/$(arg camera1_name)/color/image_raw" />
    <node name="republish_depth" type="republish" pkg="image_transport" args="compressed in:=/$(arg camera1_name)/aligned_depth_to_color/image_raw raw out:=/$(arg camera1_name)/aligned_depth_to_color/image_raw "/> -->

    <param name="use_sim_time" type="bool" value="true"/>

    <!-- Rtabmap -->
    <include file="$(find rtabmap_ros)/launch/rtabmap.launch">
        <arg name="localization"            value="$(arg localization)"/>
        <arg name="map_frame_id"            value="map_rtab"/>
        <arg name="frame_id"                value="$(arg camera1_name)_link"/> 
        <arg name="database_path"           value="$(arg db_folder)/rtabmap_ros/$(arg database_name)"/>

        <arg name="rtabmap_args"            value="--delete_db_on_start" 
            if="$(eval arg('localization') == 'false')"/> 

        <!-- Use depth -->
        <!-- <arg name="rgbd_sync"               value="true"/> 
        <arg name="depth_topic"             value="/$(arg camera1_name)/aligned_depth_to_color/image_raw"/>
        <arg name="rgb_topic"               value="/$(arg camera1_name)/color/image_raw"/>
        <arg name="camera_info_topic"       value="/$(arg camera1_name)/color/camera_info"/> -->

        <!-- Use stereo -->
        <arg name="stereo"                  value="true"/> 
        <arg name="left_image_topic"        value="/$(arg camera1_name)/infra1/image_rect_raw"/>
        <arg name="right_image_topic"       value="/$(arg camera1_name)/infra2/image_rect_raw"/>
        <arg name="left_camera_info_topic"  value="/$(arg camera1_name)/infra1/camera_info"/>
        <arg name="right_camera_info_topic" value="/$(arg camera1_name)/infra2/camera_info"/>
        
        <arg name="visual_odometry"         value="false"/>
        <arg name="odom_topic"              value="/$(arg camera2_name)/odom/sample"/>
        
        <arg name="approx_sync"             value="true"/>
        <arg name="queue_size"              value="200"/>
        <arg name="rviz"                    value="false"/>
        <arg name="rtabmapviz"              value="false"/>

    </include>


    <!-- rviz for visualization -->
    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find neptune_ros)launch/slam/visualize.rviz"/>

    <!-- Create point cloud for the local planner -->
    <!-- <node pkg="nodelet" type="nodelet" name="obstacles_detection" args="load rtabmap_ros/obstacles_detection stereo_nodelet">
        <remap from="/$(arg camera1_name)/cloud" to="/voxel_cloud"/>
        <remap from="/$(arg camera1_name)/obstacles" to="/planner_cloud"/>

        <param name="frame_id" type="string" value="$(arg camera1_name)_link"/>
        <param name="map_frame_id" type="string" value="map"/>
        <param name="min_cluster_size" type="int" value="20"/>
        <param name="max_obstacles_height" type="double" value="0.0"/>
    </node> -->

</launch>